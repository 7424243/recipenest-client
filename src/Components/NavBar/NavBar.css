nav {
    margin-left: 50px;
    margin-right: 50px;
    font-family: 'Baskervville', serif;
    font-size: medium;
}


/* @media */

@media screen and (min-width: 711px) {
    nav {
        margin-right: 50px;
        margin-top: 40px;
    }
}