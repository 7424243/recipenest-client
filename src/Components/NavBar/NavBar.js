import React, {Component} from 'react'
import { Link } from 'react-router-dom'
import TokenService from '../../services/token-service'
import './NavBar.css'

class NavBar extends Component {

    logoutLink() {
        return (
            <Link to='/' className='nav-link'> Logout </Link>
        )
    }

    loginLink() {
        return (
            <Link to='/login' className='nav-link'> Login </Link>
        )
    }
    
    render() {
        return (
            <nav>
                <Link to='/recipes' className='nav-link'>All Recipes |</Link>
                <Link to='/recipes' className='nav-link'> My Recipes |</Link>
                {TokenService.hasAuthToken() ? this.logoutLink() : this.loginLink()}
                
            </nav>
        )
    }

}

export default NavBar