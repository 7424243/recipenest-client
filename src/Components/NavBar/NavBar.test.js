import React from 'react'
import ReactDOM from 'react-dom'
import {BrowserRouter} from 'react-router-dom'
import renderer from 'react-test-renderer'
import NavBar from './NavBar';

describe('NavBar Component', () => {

  //smoke test
  it('renders without crashing', () => {
    const div = document.createElement('div')
    ReactDOM.render(<BrowserRouter><NavBar/></BrowserRouter>, div)
    ReactDOM.unmountComponentAtNode(div)
  })

  //snapshot test
  it('renders the UI as expected', () => {
    const tree = renderer 
      .create(<BrowserRouter><NavBar/></BrowserRouter>)
      .toJSON()
    expect(tree).toMatchSnapshot()
  })
  
})
