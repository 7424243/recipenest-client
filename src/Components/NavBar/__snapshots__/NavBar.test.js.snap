// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NavBar Component renders the UI as expected 1`] = `
<nav>
  <a
    href="/recipes"
    onClick={[Function]}
  >
    All Recipes |
  </a>
  <a
    href="/recipes"
    onClick={[Function]}
  >
     My Recipes |
  </a>
  <a
    href="/login"
    onClick={[Function]}
  >
     Login 
  </a>
</nav>
`;
